uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/ChangeRequestServiceImpl.java

```diff
@@ -1333,4 +1334,61 @@ JGitPathImpl createJGitPathImpl(final JGitFileSystem fs,
                                    objectId,
                                    isRealPath);
     }
-}
\ No newline at end of file
+
+    @Override
+    public List<ChangeRequestCommit> getCommits(final String spaceName,
+                                                final String repositoryAlias,
+                                                final Long changeRequestId) {
+
+        final Repository repository = resolveRepository(spaceName,
+                                                        repositoryAlias);
+
+        final ChangeRequest changeRequest = getChangeRequestById(spaceName,
+                                                                 repositoryAlias,
+                                                                 false,
+                                                                 changeRequestId);
+
+        final Git git = getGitFromBranch(repository,
+                                         changeRequest.getSourceBranch());
+
+        final String endCommitId = git.getLastCommit(changeRequest.getSourceBranch())
+                                      .getName();
+
+        final String startCommitId = changeRequest.getStartCommitId();
+
+        return git.listCommits(startCommitId,
+                               endCommitId)
+                  .stream()
+                  .map(c -> new ChangeRequestCommit(c.getName(),
+                                                    c.getFullMessage()))
+                  .collect(Collectors.toList());
+    }
+
+    @Override
+    public Boolean squashChangeRequest(final String spaceName,
+                                    final String repositoryAlias,
+                                    final Long changeRequestId,
+                                    final String commitMessage) {
+
+        final Repository repository = resolveRepository(spaceName,
+                                                        repositoryAlias);
+
+        final ChangeRequest changeRequest = getChangeRequestById(spaceName,
+                                                                 repositoryAlias,
+                                                                 false,
+                                                                 changeRequestId);
+
+        final Git git = getGitFromBranch(repository,
+                                         changeRequest.getSourceBranch());
+
+        final String startCommit = changeRequest.getStartCommitId();
+
+        git.squash(changeRequest.getSourceBranch(),
+                   startCommit,
+                   commitMessage);
+
+        return mergeChangeRequest(spaceName,
```

Comment:
Question: what happens after the squash+merge if there is only one commit?

Reply:
works fine, I could not see any edge case here

Reply:
After merging one commit and trying to do squash , getting no response from page.Upon refreshing and navigating to Change request  getting error as "The source and target branches are in conflict, thus cannot be integrated."
But same squash commit message can be seen under version.


![After 1st merge and squash error](https://user-images.githubusercontent.com/57942961/72064140-4000ce80-3301-11ea-984d-4addc61188ca.png)
![After 1st merge and squash under version](https://user-images.githubusercontent.com/57942961/72064153-48590980-3301-11ea-9b6e-c81006e6a4bd.png)

Reply:
Hi @Srihari1192 
Thank you for catching this up
I could not reproduce the scenario you are describing. Can you provide more details ?
Merge + Squash a single commit works fine for me

Reply:
Steps:
1.add asset data object.
2.Submit change request.
3.accept the change request with master(Note: not doing squash just merge).
4.Again modify asset and create change request.
5.Now try to squash the changes  with master and after click on 'Ok', not showing any response
6.After step 5-> we are able to see squash message under Version but change request is prompting still has "The source and target branches are in conflict, thus cannot be integrated.".
So two observations step 5 and step 6..Please refer screenshot for reference
![Step1](https://user-images.githubusercontent.com/57942961/72166795-eaefb600-33ef-11ea-95f7-931fe9052380.png)
![Step2](https://user-images.githubusercontent.com/57942961/72166799-eb884c80-33ef-11ea-8d6a-3c4b65b6e109.png)
![Step3](https://user-images.githubusercontent.com/57942961/72166800-ecb97980-33ef-11ea-9702-9c6a0ecd86b2.png)
![Step4](https://user-images.githubusercontent.com/57942961/72166806-efb46a00-33ef-11ea-8ca0-90dad6abf210.png)
![Step5](https://user-images.githubusercontent.com/57942961/72166808-f04d0080-33ef-11ea-8c9f-4ec08f9ed25e.png)
![Step6](https://user-images.githubusercontent.com/57942961/72166810-f04d0080-33ef-11ea-8883-f8d82cb43277.png)
![Step6_1](https://user-images.githubusercontent.com/57942961/72166812-f0e59700-33ef-11ea-81e7-e724f22d1cfe.png)
![Step 4_1](https://user-images.githubusercontent.com/57942961/72166814-f17e2d80-33ef-11ea-966e-cf31d71d6f5f.png)

Reply:
I submit a new commit to solve this issue
Let me know if this works

Reply:
@gabriel376 Now during creation of change request getting error as "The source and target branches are in conflict!" but same we can able to perform successfully, if we do before merge and try to do multiple merge.
Steps:
1.Add asset with created branch.
2.Create change request and Squash the commit.
3.Again modify asset and try to create change request.
4.getting error as "The source and target branches are in conflict!" .

Note:Same steps is getting successful, if we perform only merge actions.